### PR TITLE
Buildfix and cleanup

### PR DESCRIPTION
**Pull request by Kicer86**

### CODE REVIEW

File: include/exiv2/asfvideo.hpp [hunk 1]
```diff
@@ -47,7 +46,7 @@ namespace Exiv2 {
     /*!
       @brief Class to access ASF video files.
      */
-    class EXIV2_DEPRECATED EXIV2API AsfVideo:public Image
+    class EXIV2LIB_DEPRECATED_EXPORT AsfVideo:public Image
```
Comment (D4N) [inline include/exiv2/asfvideo.hpp]: Where does this macro come from?

Comment (Kicer86) [reply include/exiv2/asfvideo.hpp]: exiv2lib_export.h

Comment (D4N) [reply include/exiv2/asfvideo.hpp]: Ah, thanks. But then, how does this help with your problem? The macro `EXIV2LIB_DEPRECATED_EXPORT` is defined as `EXIV2API EXIV2_DEPRECATED` (on my machine), so unless you want to forcefully redefine it, won't you run into the same problem with the class with the `__attribute__` and `[]` syntax?

**Review by piponazo — APPROVED** [review]
Thanks for the contribution! I learned something new about CMake with this PR 😉 

I have the feeling that with this change, we might be able to add back the VS2008 job in Appveyor. I'll try that once the PR is merged.

**Comment by Kicer86** [comment]
:)

File: include/exiv2/asfvideo.hpp (hunk 1 above)
Comment (Kicer86) [reply include/exiv2/asfvideo.hpp]: @D4N  Luckly both `EXIV2API`  and `EXIV2_DEPRECATED` always expand into `__attribute__` form, so there won't be a mix of `__attribute__` and `[[]]`

**Pull request merged**